pkg/server/build/build.go

```diff
@@ -0,0 +1,76 @@
+package build
+
+import (
+	"fmt"
+	"io"
+
+	context "golang.org/x/net/context"
+
+	"github.com/luizalabs/teresa/pkg/server/app"
+	"github.com/luizalabs/teresa/pkg/server/exec"
+	"github.com/luizalabs/teresa/pkg/server/spec"
+	"github.com/luizalabs/teresa/pkg/server/storage"
+)
+
+type Operations interface {
+	CreateByOpts(ctx context.Context, opt *CreateOptions) error
+}
+
+type Options struct {
+	SlugBuilderImage string
+	BuildLimitCPU    string
+	BuildLimitMemory string
+}
+
+type BuildOperations struct {
+	fileStorage storage.Storage
+	execOps     exec.Operations
+	opts        *Options
+}
+
+// CreateOptions define arguments of method `CreateByOpts`
+type CreateOptions struct {
+	App       *app.App
+	BuildName string
+	SlugIn    string
+	SlugDest  string
+	TarBall   io.ReadSeeker
+	Stream    io.Writer
+}
+
+func (ops BuildOperations) CreateByOpts(ctx context.Context, opt *CreateOptions) error {
```

Comment:
My suggestions regarding build.go: CreateFromOpts, always use `opts` for a pointer to Options and use Operations instead of BuildOperations.

Reply:
LGTM about `opts`.
About `BuildOperations`, BuildOperations is the concrete implementation and Operations is the Interface, what do you think?